integration-tests/src/tests/client/sharding_upgrade.rs

```diff
@@ -166,21 +241,21 @@ impl TestShardUpgradeEnv {
     /// layout V2 to be used once the appropriate protocol version is reached
     fn step_impl(
         &mut self,
-        p_drop_chunk: f64,
+        drop_chunk_condition: &DropChunkCondition,
         protocol_version: ProtocolVersion,
         resharding_type: &ReshardingType,
     ) {
         let env = &mut self.env;
         let mut rng = thread_rng();
```

Comment:
Yep, let me follow up and implement this in a separate PR. FYI this test is randomized on many different levels and I don't think it's possible to fix all of them but I'll try to cover most of it. I also think there is value in having the test run on different random seeds every time. Rather than making it deterministic I would aim to make it reproducible by picking the seed at random and printing it to stdout. This way we can reproduce the issue when debugging by hardcoding the seed to a known bad value but we still get the randomized coverage.

Reply:
> I also think there is value in having the test run on different random seeds every time. 

I would argue that there is not much value in that. If you want more coverage then just run the test with more data/samples. Having reproducible tests helps to avoid flakiness, which is a real issue in our codebase.

Reply:
Fair enough, let me get the randomness to be seeded from a single place first and then we can continue the discussion.